src/main/scala/rps/RPSMoves.scala

```diff
@@ -0,0 +1,37 @@
+package rps
+
+
+/*
+
+Scala has a strange view of Enumerations, that is
+a little bit too much for the average use-case.
+
+The "agile-version" of enumerations are called
+"sealed trait extended by case objects".
+
+Here we define a RPSMoves ADT that will contain
+useful definitions and methods to craft RPS Moves.
+
+*/
+
+
+object RPSMoves {
```

Comment:
The general convention is to define a class as toplevel and then use a [companion object](https://docs.scala-lang.org/tour/singleton-objects.html#companion-objects) for adding static members and namepsacing. In this case, this would look like

```scala
sealed abstract class RPSMove(val icon: String, val value: String)
object RPSMove {
  case object Rock extends RPSMove("💎", "0")
  // ...
}
```

Reply:
Gotcha! Does this way of writing "agile-enumerators" with a companion object and an abstract class as toplevel has a nickname?

Reply:
This is the most common encoding in Scala of a "sum type" (also known in literature as a "coproduct" or "tagged union").

This a specific case of a "sum type" where all members of the type are a single value (i.e. an `object`).

The use of `abstract class` (or `trait`) and the companion object is just practical in Scala, I don't think it has a name.

Reply:
Perfect! `Sum type`. Better know the nomenclature right from the start!

Reply:
Another name for this is `Algebraic Data Type`, or `ADT` (not to be confused with `Abstract Data Type`)